packages/gatsby-source-shopify/src/query-builders/product-variants-query.ts

```diff
@@ -0,0 +1,90 @@
+import { BulkQuery } from "./bulk-query"
+
+export class ProductVariantsQuery extends BulkQuery {
+  query(date?: Date): string {
+    const publishedStatus = this.pluginOptions.salesChannel
+      ? encodeURIComponent(`${this.pluginOptions.salesChannel}=visible`)
+      : `published`
+
+    const filters = [`status:active`, `published_status:${publishedStatus}`]
+    if (date) {
+      const isoDate = date.toISOString()
+      filters.push(`created_at:>='${isoDate}' OR updated_at:>='${isoDate}'`)
+    }
+
+    const ProductVariantSortKey = `POSITION`
+
+    const queryString = filters.map(f => `(${f})`).join(` AND `)
+
+    const query = `
+      {
+        productVariants(query: "${queryString}", sortKey: ${ProductVariantSortKey}) {
+          edges {
+            node {
+              availableForSale
+              barcode
+              compareAtPrice
+              createdAt
+              displayName
+              id
+              image {
+                id
+                altText
+                height
+                width
+                originalSrc
+                transformedSrc
+              }
+              inventoryPolicy
+              inventoryQuantity
+              legacyResourceId
+              position
+              price
+              selectedOptions {
+                name
+                value
+              }
+              sellingPlanGroupCount
+              sku
+              storefrontId
+              taxCode
+              taxable
+              title
+              updatedAt
+              weight
+              weightUnit
+              presentmentPrices {
+                compareAtPrice {
+                  amount
+                  currencyCode
+                }
+                price {
+                  amount
+                  currencyCode
+                }
```

Comment:
This will have to be removed, it's only in here right now for testing. When this PR is merged we have to follow-up to add this back in, and add custom processing logic for variants that will attach the `presentmentPrices` to variants as a simple array. This is necessary because `presentmentPrices` don't have node IDs, so we can't create nodes from them.